Article/Article.js

```diff
@@ -1,20 +1,25 @@
 // Because classes are not hoisted you will need to start your code at the bottom of the page.  Look for the comment "START HERE"
 
 class Article {
-  constructor(domElement) {
+  constructor(article) {
     // assign this.domElement to the passed in domElement
-    this.domElement;
+    this.article = article;
     // create a reference to the ".expandButton" class. 
-    this.expandButton;
+    this.expandButton = this.article.querySelector('.expandButton');
     // Using your expandButton reference, update the text on your expandButton to say "expand"
-    
+    this.expandButton.textContent = 'expand';
     // Set a click handler on the expandButton reference, calling the expandArticle method.
-    
+    this.expandButton.addEventListener('click', () => this.expandArticle());
+    // adding hide button for stretch
+    this.retractButton = this.article.querySelector('.retractButton');
+    this.retractButton.textContent = 'hide';
+    this.retractButton.addEventListener('click', () => this.expandArticle());
   }
-
   expandArticle() {
     // Using our reference to the domElement, toggle a class to expand or hide the article.
-
+    this.article.classList.toggle('article-open');
+    this.expandButton.classList.toggle('hide-btn');
+    this.retractButton.classList.toggle('hide-btn');
```

Comment:
Nice idea with toggling a `"hide"` class to change the button for opening and closing an article!

You don't actually have to use two separate button elements, though - another idea would be to simply change the `textContent` property of your expand button (since it never actually disappears; rather, the text just changes).